src/cdk/tree/tree.ts

```diff
@@ -328,23 +334,30 @@ export class CdkTreeNode<T> implements FocusableOption, OnDestroy {
   }
   protected _data: T;
 
-  get isExpanded(): boolean {
+  // tslint:disable-next-line:no-host-decorator-in-concrete
+  @HostBinding('attr.aria-expanded') get isExpanded(): boolean {
     return this._tree.treeControl.isExpanded(this._data);
   }
 
   get level(): number {
-    return this._tree.treeControl.getLevel ? this._tree.treeControl.getLevel(this._data) : 0;
-  }
-
-  /**
-   * The role of the node should always be 'treeitem'.
-   */
-  // TODO: mark as deprecated
-  @Input() role: 'treeitem' | 'group' = 'treeitem';
+   // If the treeControl has a getLevel method, use it to get the level. Otherwise read the
+   // aria-level off the parent node and use it as the level for this node (note aria-level is
+   // 1-indexed, while this property is 0-indexed, so we don't need to increment).
+   return this._tree.treeControl.getLevel ?
+     this._tree.treeControl.getLevel(this._data) : this._parentNodeAriaLevel;
+   }
 
   constructor(protected _elementRef: ElementRef<HTMLElement>,
               protected _tree: CdkTree<T>) {
     CdkTreeNode.mostRecentTreeNode = this as CdkTreeNode<T>;
+    // The classes are directly added here instead of in the host property because classes on
+    // the host property are not inherited with View Engine.
+    this._elementRef.nativeElement.classList.add('cdk-tree-node');
+  }
+
+  ngOnInit(): void {
+    this._parentNodeAriaLevel = getParentNodeAriaLevel(this._elementRef.nativeElement);
+    this._elementRef.nativeElement.setAttribute('aria-level', String(this.level + 1));
```

Comment:
I think we generally convert things to string like: `this.level + 1 + ''`.

Reply:
I usually convert things to strings like this: `` `${this.level + 1}` ``